### PR TITLE
Attempt to define "active cohort member"

### PR DESCRIPTION
**Pull request by ibacher**
@corneliouzbett Interested in your take on this. I'm not too sure about this, but it seems like a nice-to-have to have an easy way of getting the active cohort members.

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/cohort/CohortM.java [hunk 1]
```diff
@@ -66,7 +68,11 @@
 	private CohortType cohortType;
 	
 	@OneToMany(mappedBy = "cohort", cascade = CascadeType.ALL)
-	private List<CohortMember> cohortMembers = new ArrayList<>();
+	private Set<CohortMember> cohortMembers = new HashSet<>();
+	
+	@OneToMany(mappedBy = "cohort", cascade = CascadeType.ALL)
+	@Where(clause = "voided = 0 and (start_date is null or start_date <= current_timestamp()) and (end_date is null or end_date >= current_timestamp())")
+	private Set<CohortMember> activeCohortMembers;
```
Comment (kipchubett) [inline api/src/main/java/org/openmrs/module/cohort/CohortM.java:75]: Wouldn't this be slow for a large cohort(many members)?  Maybe adding an Entity graph for the association might improve the performance?

**Review by kipchubett — APPROVED** [review]
This LGTM!

File: api/src/main/java/org/openmrs/module/cohort/CohortM.java (hunk 1 above)
Comment (ibacher) [reply api/src/main/java/org/openmrs/module/cohort/CohortM.java:75]: Are entity graphs available in the version of Hibernate we're targetting (4.3.9)? I was under the impression they weren't.

I tried to mitigate this with an index, but that needs to be changed.

Comment (kipchubett) [reply api/src/main/java/org/openmrs/module/cohort/CohortM.java:75]: Yes. It's available. Tried to use once didn't suit the use case.

**Comment by ibacher** [comment]
@corneliouzbett For now I'm just going to merge this in. I couldn't quite get the entity graph working.

**Pull request merged**